src/zimscraperlib/zim/creator.py

```diff
@@ -146,7 +150,31 @@ def config_indexing(
         self.__indexing_configured = True
         return self
 
+    def _maybe_metadata_from_maybe_illustration(self, name, value):
```

Comment:
Lot's of `maybe`. Something like `_get_illustration_metadata` is sufficient, the fact that it might return None is self-explanatory.

Please also add a docstring explaining this function purpose.

Add also add type hints to arguments and return value